Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/workbench/RetrieveProjectManager.java

```diff
@@ -197,4 +203,72 @@ public boolean getPrivateProjectWebsiteAccessibility(String unixName, String use
 		}
 		return false;
 	}
+	
+	
+	/**
+	 * This method returns the list of recently updated top 5 projects associated with
+	 * the logged in user.
+	 * @param sUserName
+	 * @return List - list of recently updated projects.
+	 * @throws QuadrigaStorageException
+	 * @author Suraj Nilapwar
+	 */
+	@Override
+	@Transactional
+	public List<IProject> getRecentProjectList(String sUserName) throws QuadrigaStorageException
+	{
+		List<IProject> projectsList = new ArrayList<IProject>();
+		List<String> projectIds = new ArrayList<String>();
+		
+		List<IProject> projectListAsOwner;
+		projectListAsOwner =  projectShallowMapper.getProjectList(sUserName);
+		if(projectListAsOwner != null)
+		{
+			for (IProject p : projectListAsOwner) {
+				projectsList.add(getProjectDetails(p.getProjectId()));
+				projectIds.add(p.getProjectId());
+			}
+		}
+		List<IProject> projectListAsCollaborator; 
+		projectListAsCollaborator = projectShallowMapper.getCollaboratorProjectListOfUser(sUserName);;
+		if(projectListAsCollaborator != null)
+		{
+			for (IProject p : projectListAsCollaborator) {
+				if (!projectIds.contains(p.getProjectId()))
+				{
+					projectsList.add(getProjectDetails(p.getProjectId()));		
+					projectIds.add(p.getProjectId());
+				}
+			}
+		}
+		Collections.sort(projectsList, new Comparator<IProject>() {
+			@Override
+			public int compare(IProject o1, IProject o2) {
+				return o2.getUpdatedDate().compareTo(o1.getUpdatedDate());
+			}
+		});
+		
+		int count;
+		String recentProjectCount = env.getProperty("project.recent.list.count");
+		if(recentProjectCount!=null)
+		{
+			count = Integer.parseInt(recentProjectCount);
+		}
+		else
+		{
+			count = projectsList.size();
+		}
+		
+		List<IProject> recentProjectsList;
+		if(projectsList!=null && recentProjectCount!=null && projectsList.size()>count)
+		{
+			recentProjectsList = new ArrayList<IProject>(projectsList.subList(0, count));
+		}
+		else
+		{
+			recentProjectsList = new ArrayList<IProject>(projectsList);
+		}
+		
+		return recentProjectsList;
+	}

```

Comment:
Please read and follow the style guide we use (not necessarily the indentation level for this class since it's already using 6 spaces but everything else like bracket placement).